src/test/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension_PER_CLASS_Injection_Test.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api.junit.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.assertj.core.api.BDDSoftAssertions;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+@ExtendWith(SoftAssertionsExtension.class)
+@DisplayName("SoftAssertionsExtension PER_CLASS injection test")
+@TestInstance(Lifecycle.PER_CLASS)
+@Execution(ExecutionMode.SAME_THREAD) // Just in case we move to parallel threads in the future
+class SoftAssertionsExtension_PER_CLASS_Injection_Test {
+
+  @InjectSoftAssertions
+  SoftAssertions softly;
+
+  @InjectSoftAssertions
+  private BDDSoftAssertions deftly;
+  
+  @InjectSoftAssertions
+  ThreadLocal<CustomSoftAssertions> surreptitiously;
+  
+  SoftAssertions unannotated;
```

Comment:
Originally they were all package-private because I had the restriction on `private`. When I lifted the restriction on `private` I changed one of them to make sure I was covering that part of the code.